### DISCUSSION TITLE
How to mock the SDK?

### DISCUSSION THREAD

**Discussion by nmoutschen** [Byron/google-apis-rs#342] [Q&A] [answered]
Hi! 👋 

Is there a way to mock this SDK by providing expected requests/responses for unit tests, like [you can do with the AWS SDK](https://github.com/awslabs/aws-sdk-rust/blob/main/sdk/kms/tests/integration.rs#L24-L49)? For context, I'm building an integration with Google Workflows and would like to mock the requests/reponses to a certain extent while developping locally.

I know that I could alternatively write a trait/struct on top of the SDK, but that would defeat some of the purposes of my tests.

**Comment by Byron** [comment, marked as answer]
There was a way of doing that using `yup-hyper-mock`, but I doubt it still works after the last breaking update the google crates received. Maybe the mentioned crate can serve as a basis for exploring your own solution to this, maybe it even still works or is just one PR away from getting there.
